includes/class-newspack-popups-inserter.php

```diff
@@ -55,87 +55,49 @@ public static function popups_for_post() {
 			return [ Newspack_Popups_Model::retrieve_preview_popup( Newspack_Popups::previewed_popup_id() ) ];
 		}
 
-		// Campaigns disabled for this page.
+		// Popups disabled for this page.
 		if ( self::assess_has_disabled_popups() ) {
 			return [];
 		}
 
 		$view_as_spec             = Segmentation::parse_view_as( Newspack_Popups_View_As::viewing_as_spec() );
 		$view_as_spec_groups      = isset( $view_as_spec['groups'] ) ? $view_as_spec['groups'] : false;
-		$view_as_spec_all         = ! empty( $view_as_spec['all'] ) ? true : false;
-		$view_as_spec_segment     = isset( $view_as_spec['segment'] ) ? $view_as_spec['segment'] : false;
 		$view_as_spec_unpublished = isset( $view_as_spec['show_unpublished'] ) && 'true' === $view_as_spec['show_unpublished'] ? true : false;
 
-		if ( $view_as_spec_groups ) {
-			// If previewing specific groups.
-			$popups_to_maybe_display = Newspack_Popups_Model::retrieve_group_popups( explode( ',', $view_as_spec['groups'] ), $view_as_spec_unpublished );
-		} elseif ( $view_as_spec_all || $view_as_spec_segment ) {
-			// If previewing and no groups are specified, but 'all' or a segment is specified, retrieve all campaigns.
-			$popups_to_maybe_display = Newspack_Popups_Model::retrieve_popups( $view_as_spec_unpublished );
-		} else {
-			// Retrieve campaigns for front-end display.
+		// Retrieve all popups eligible for display.
 
-			// 1. Get all inline popups.
-			$popups_to_maybe_display = Newspack_Popups_Model::retrieve_inline_popups();
+		// 1. If previewing specific groups, only get popups that match those groups.
+		$group_slugs = $view_as_spec_groups ? explode( ',', $view_as_spec['groups'] ) : false;
 
-			// Check if there's an overlay popup with matching category.
-			$category_overlay_popup = Newspack_Popups_Model::retrieve_category_overlay_popup();
-			if ( $category_overlay_popup && self::should_display( $category_overlay_popup ) ) {
-				array_push(
-					$popups_to_maybe_display,
-					$category_overlay_popup
-				);
-			} else {
-				// If there's no category-matching popup, get the sitewide pop-up.
-				$sitewide_default = get_option( Newspack_Popups::NEWSPACK_POPUPS_SITEWIDE_DEFAULT, null );
-				if ( $sitewide_default ) {
-					$found_popup = Newspack_Popups_Model::retrieve_popup_by_id( $sitewide_default );
-					if (
-						$found_popup &&
-						// Prevent non-overlay sitewide default from being added.
-						Newspack_Popups_Model::is_overlay( $found_popup )
-					) {
-						array_push(
-							$popups_to_maybe_display,
-							$found_popup
-						);
-					}
-				}
-			}
-		}
+		// 2. Get all inline and above-header popups.
```

Comment:
In other places in the code and in the UI, an "above header" popup _is_ an inline one:

![image](https://user-images.githubusercontent.com/7383192/106306008-ba2e2900-625d-11eb-8ce5-0cbd8e7078f8.png)

```suggestion
		// 2. Get all inline popups.
```

Reply:
True, but we're treating above-headers differently than other inlines in the code. Still, I get what you're saying.